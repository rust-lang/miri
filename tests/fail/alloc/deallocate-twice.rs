use std::alloc::{alloc, dealloc, Layout};

//@error-pattern: dereferenced after this allocation got freed

fn main() {
    unsafe {
        let x = alloc(Layout::from_size_align_unchecked(1, 1));
        dealloc(x, Layout::from_size_align_unchecked(1, 1));
        dealloc(x, Layout::from_size_align_unchecked(1, 1));
    }
}
