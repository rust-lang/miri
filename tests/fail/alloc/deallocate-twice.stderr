error: Undefined Behavior: pointer to ALLOC was dereferenced after this allocation got freed
  --> RUSTLIB/alloc/src/alloc.rs:LL:CC
   |
LL |     unsafe { __rust_dealloc(ptr, layout.size(), layout.align()) }
   |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ pointer to ALLOC was dereferenced after this allocation got freed
   |
   = help: this indicates a bug in the program: it performed an invalid operation, and caused Undefined Behavior
   = help: see https://doc.rust-lang.org/nightly/reference/behavior-considered-undefined.html for further information
   = note: BACKTRACE:
   = note: inside `std::alloc::dealloc` at RUSTLIB/alloc/src/alloc.rs:LL:CC
note: inside `main` at $DIR/deallocate-twice.rs:LL:CC
  --> $DIR/deallocate-twice.rs:LL:CC
   |
LL |         dealloc(x, Layout::from_size_align_unchecked(1, 1));
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

note: some details are omitted, run with `MIRIFLAGS=-Zmiri-backtrace=full` for a verbose backtrace

error: aborting due to previous error

